compaction/compaction_descriptor.hh

```diff
@@ -171,6 +171,9 @@ struct compaction_descriptor {
     // Denotes if this compaction task is comprised solely of completely expired SSTables
     sstables::has_only_fully_expired has_only_fully_expired = has_only_fully_expired::no;
 
+    // If enabled, gc will check only the compacting sstables to collect tombstones
+    bool gc_check_only_compacting_sstables = false;
+
```

Comment:
> Could/should we make this more accurate? Instead of checking a flag (which could be correct/incorrect), check timestamps in memtable/commitlog/new sstables and if they're in the future, ignore them.

We already do that but in a very inaccurate way that generates false-positives (per memtable write timestamp ranges IIUC, not even per-partition, nor per-row).

> 
> I guess that doesn't work for shadowable timestamps, does it? Maybe a comment here to explain the situation.

right, with shadowable tombstone we need to check row-marker tombstones that may shadow the shadowable tombstone.

Reply:
Latest version adds more information about what is being checked and how.